src/lib/Microsoft.Fx.Portability.Cci/Analyzer/DependencyFinderEngine.cs

```diff
@@ -60,22 +60,30 @@ private void FindDependencies(IProgressTask progressTask)
                 {
                     foreach (var dep in GetDependencies(filename))
                     {
-                        MemberInfo m = new MemberInfo() { MemberDocId = dep.MemberDocId, TypeDocId = dep.TypeDocId, DefinedInAssemblyIdentity = dep.DefinedInAssemblyIdentity };
+                        var m = new MemberInfo
+                        {
+                            MemberDocId = dep.MemberDocId,
+                            TypeDocId = dep.TypeDocId,
+                            DefinedInAssemblyIdentity = dep.DefinedInAssemblyIdentity
+                        };
 
                         // Add this memberinfo
                         var newassembly = new HashSet<AssemblyInfo>
                         {
                             dep.CallingAssembly
                         };
-                        ICollection<AssemblyInfo> assemblies = dependencies.AddOrUpdate(m, newassembly, (key, existingSet) =>
-                                                                                                        {
-                                                                                                            lock (existingSet)
-                                                                                                            {
-                                                                                                                existingSet.Add(dep.CallingAssembly);
-                                                                                                            }
-                                                                                                            return existingSet;
-                                                                                                        });
+
+                        var assemblies = dependencies.AddOrUpdate(m, newassembly, (key, existingSet) =>
+                        {
+                            lock (existingSet)
```

Comment:
Consider removing this lock. ConcurrentDictionary ensures that updates to the values done in the factories are "atomic".

Reply:
I would prefer to take care of these in another pass.  Open https://github.com/microsoft/dotnet-apiport/issues/810 to track the issue.